### ISSUE TITLE
Spec out an effect system

### ISSUE THREAD

**Issue by brouberol** [brouberol/5esheets#77]
The goal here is to come up with a proof of concept or a spec of a system that would allow users to define effects, whether they would apply on items, spells equipment or something else, and have that effect be evaluated at runtime.

For example, we could define an armour with an effect of `add +3 to your AC`, and how the AC is calculated, its score, etc, will all be dynamically evaluated. 

Let's figure out the syntax of that system, and its limits

**Comment by brouberol** [comment]
Re transcribing some messages we exchanged on the subject:

> For effects, I figured we could find some kind of DSL that works for all relationships.
> We could write `strength_save_mod = Math.floor(strength / 2 + proficiency)` and it automatically adds the right effect, saving the dependencies. And for objects we could create an effect `armor_class = base_armor_class + dex`.
> What I had in mind looks a lot like foundry's active effects: https://foundryvtt.com/article/active-effects/
> There are 5 modes, for which we would have 5 operators:
```
  Add: +=
  Multiply: *=
  Override:=
  Downgrade: >=
  Upgrade: <=
  ```
> And then you can use a special sign (e.g. _, @, $) to access the properties of the character sheet, and compose them.

### About modes

Here's a copy/paste of the [Foundry documentation](https://foundryvtt.com/article/active-effects/) about effect modes, for reference


**Add** LAdds the provided value to a number. This can be used to both add and subtract from a particular value by specifying `+1` or `-1` as the value to add.

**Multiply**: Multiplies the defined attribute by the numeric value in the Effect Value field.

**Override**: Replaces the defined attribute with the value provided in the Effect Value field.

**Downgrade**: Reduces the defined attribute only in cases where the current value of that attribute would be greater than value specified in the Effect Value field.

**Upgrade**: Increases the defined attribute only in cases where the current value of that attribute would be less than value specified in the Effect Value field.

**Comment by brouberol** [comment]
Fixed by #82
